### PR TITLE
[WIP] New Nomad Vault Integration Guide

### PR DESCRIPTION
**Pull request by Omar-Khawaja**
Moves existing Vault integration content from the guides section to the docs section. There is now a new Vault integration guide in the guides section (in the same spot as the previous Vault integration content).

### CODE REVIEW

**Comment by Omar-Khawaja** [comment]
@rcgenova let me know how moving the old Vault integration content to the docs section looks. The new guide (not 100% finished) will take its place in the guides section.

File: website/source/docs/vault-integration/index.html.md [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+```
+# Download the token role
+$ curl https://nomadproject.io/data/vault/nomad-cluster-role.json -O -s -L
+
+# Create the token role with Vault
+$ vault write /auth/token/roles/nomad-cluster @nomad-cluster-role.json
+```
+
+
+#### Example Configuration
```
Comment (yhyakuna) [inline website/source/docs/vault-integration/index.html.md]: This section is repetitive and seems unnecessary to me.

File: website/source/docs/vault-integration/index.html.md [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+$ vault write /auth/token/roles/nomad-cluster @nomad-cluster-role.json
+```
+
+#### Retrieving the Token Role based Token
+
+After the token role is created, a token suitable for the Nomad servers may be
+retrieved by issuing the following Vault command:
+
+```
+$ vault token create -policy nomad-server -period 72h -orphan
```
Comment (yhyakuna) [inline website/source/docs/vault-integration/index.html.md]: If you are generating a token based on a defined role, the command should be:
`vault token create -role=nomad-cluster`

Otherwise, the previous step to create a role becomes meaningless.

File: website/source/docs/vault-integration/index.html.md [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+the list. If `disallowed_policies` is used, task may request any policy that is
+not in the `disallowed_policies` list. There are trade-offs to both approaches
+but generally it is easier to use the blacklist approach and add policies that
+you would not like tasks to have access to into the `disallowed_policies` list.
+
+An example token role definition is given below:
+
+```json
+{
+  "disallowed_policies": "nomad-server",
```
Comment (yhyakuna) [inline website/source/docs/vault-integration/index.html.md]: Did you really mean to *disallow* `nomad-server` policy, or you meant to specify `allowed_policies`?

Comment (Omar-Khawaja) [reply website/source/docs/vault-integration/index.html.md]: /guides/operations/vault-integration/index.html

File: website/source/docs/vault-integration/index.html.md (hunk 1 above)
Comment (Omar-Khawaja) [reply website/source/docs/vault-integration/index.html.md]: New guide located at /guides/operations/vault-integration/index.html

File: website/source/guides/operations/vault-integration/index.html.md [hunk 4]
```diff
@@ -145,207 +209,293 @@ An example token role definition is given below:
...
+{
+  "plugin_name": "postgresql-database-plugin",
+  "allowed_roles": "accessdb",
+  "connection_url": "postgresql://{{username}}:{{password}}@database.service.consul:5432/postgres?sslmode=disable",
+  "username": "postgres",
+  "password": "postgres123"
 }
 ```
+The information above allows Vault to connect to our database and create users
+with certain privileges. We will specify the `accessdb` role soon. Please note that it is very common to give Vault the root credentials and let Vault manage the auditing and lifecycle of generated credentials.
```
Comment (yhyakuna) [inline website/source/guides/operations/vault-integration/index.html.md]: It's my fault and I need to update the Vault doc, but please remove **root credentials** from the last sentence.  So, rephrase it to say something like: 
`Note that Vault needs enough privilege to generate database credentials dynamically and manage their lifecycle.`

File: website/source/guides/operations/vault-integration/index.html.md [hunk 5]
```diff
@@ -1,71 +1,124 @@
...
+retrieve dynamic credentials for various tasks. In this guide, you will deploy a
+web application that needs to authenticate against [PostgreSQL][postgresql] to
+display data from a table to the user.
 
-Note that in order to use Vault with Nomad, you will need to configure and
-install Vault separately from Nomad. Nomad does not run Vault for you.
+## Reference Material
 
--> **Note:** Vault integration requires Vault version 0.6.2 or higher.
+- [Vault Integration][vault-integration]
```
Comment (rcgenova) [inline website/source/guides/operations/vault-integration/index.html.md]: We may want to label this link "Vault Integration Docs Page" or similar here to differentiate it from this new Vault Integration guide/tutorial page.

File: website/source/guides/operations/vault-integration/index.html.md [hunk 6]
```diff
@@ -1,71 +1,124 @@
...
+Success! You are now authenticated. The token information displayed below
+is already stored in the token helper. You do NOT need to run "vault login"
+again. Future Vault requests will automatically use this token.
+...
+```
+### Step 4: Write the Policy for the Nomad Server Token
+
+To use the Vault integration, you must provide a Vault token to your Nomad
+servers. Although you can provide your root token to easily get started, the
+recommended approach is to use a token [role][vault-integration] based token.
```
Comment (rcgenova) [inline website/source/guides/operations/vault-integration/index.html.md]: possibly link [role] to https://www.vaultproject.io/docs/auth/token.html

File: website/source/guides/operations/vault-integration/index.html.md [hunk 7]
```diff
@@ -145,207 +209,449 @@ An example token role definition is given below:
...
+{
+  "plugin_name": "postgresql-database-plugin",
+  "allowed_roles": "accessdb",
+  "connection_url": "postgresql://{{username}}:{{password}}@database.service.consul:5432/postgres?sslmode=disable",
+  "username": "postgres",
+  "password": "postgres123"
+}
+```
+The information above allows Vault to connect to our database and create users
+with certain privileges. We will specify the `accessdb` role soon. In a
```
Comment (rcgenova) [inline website/source/guides/operations/vault-integration/index.html.md]: s/certain/specific

File: website/source/guides/operations/vault-integration/index.html.md [hunk 8]
```diff
@@ -145,207 +209,449 @@ An example token role definition is given below:
...
+production setting, it is recommended to give Vault credentials with enough
+privileges to generate database credentials dynamically and and manage their
+lifecycle.
 
-The token can then be set in the server configuration's
-[`vault` stanza][config], as a command-line flag, or via an environment
-variable.
+Run the the following command to configure the connection between the database
```
Comment (rcgenova) [inline website/source/guides/operations/vault-integration/index.html.md]: "the the"

File: website/source/guides/operations/vault-integration/index.html.md [hunk 9]
```diff
@@ -145,207 +209,450 @@ An example token role definition is given below:
...
+Use the following command to run the job:
 
+```shell
+$ nomad run web-app.nomad 
 ```
-Code: 403. Errors:
-URL: PUT https://vault:8200/v1/sys/leases/renew
-* permission denied
+
+### Step 14: Confirm the Application is Accessing the Database
```
[the remark this answers is not in the document]
Comment (Omar-Khawaja) [reply website/source/guides/operations/vault-integration/index.html.md]: Good catch! Thanks

**Comment by Omar-Khawaja** [comment]
@rcgenova took into account the fact that there were various parts of the website that pointed to the old vault integration content. I fixed the links accordingly so that they still point to the original content as intended.

https://github.com/hashicorp/nomad/pull/4886/commits/431e924dd4b113f391b141546d3c6e937b34f8b1
